app/src/main/java/com/wut/indoornavigation/data/mesh/Mesh.java

```diff
@@ -0,0 +1,234 @@
+package com.wut.indoornavigation.data.mesh;
+
+import com.wut.indoornavigation.data.graph.Graph;
+import com.wut.indoornavigation.data.graph.HeuristicFunction;
+import com.wut.indoornavigation.data.graph.UnionFind;
+import com.wut.indoornavigation.data.graph.VertexComparator;
+import com.wut.indoornavigation.data.graph.impl.GraphImpl;
+import com.wut.indoornavigation.data.model.Building;
+import com.wut.indoornavigation.data.model.Elevator;
+import com.wut.indoornavigation.data.model.Floor;
+import com.wut.indoornavigation.data.model.FloorObject;
+import com.wut.indoornavigation.data.model.Point;
+import com.wut.indoornavigation.data.model.Stairs;
+import com.wut.indoornavigation.data.model.graph.Vertex;
+import com.wut.indoornavigation.data.model.mesh.MeshResult;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public final class Mesh {
+    private static final int ID_SEED_INIT = -1;
+    private static final double HORIZONTAL_VERTICAL_EDGE_WEIGHT = 0.5;
+    private static final double DIAGONAL_EDGE_WEIGHT = 0.7;
+    private static final int EDGE_ELEVATOR_WEIGHT = 5000;
+
+    private static int idSeed;
+
+    private Map<Integer, List<Vertex>> destinationVerticesDict;
+    private Map<Integer, List<Vertex>> elevatorsVerticesDict;
+    private Map<Integer, List<Vertex>> stairsVerticesDict;
+
+    public MeshResult create(Building building) {
+        idSeed = ID_SEED_INIT;
+        destinationVerticesDict = new HashMap<>();
+        elevatorsVerticesDict = new HashMap<>();
+        stairsVerticesDict = new HashMap<>();
+
+        HeuristicFunction heuristicFunction = new HeuristicFunction();
+        UnionFind unionFind = new UnionFind();
+
+        Graph graph = new GraphImpl(heuristicFunction, unionFind, new VertexComparator(heuristicFunction));
```

Comment:
to musi byc odswieżane przy kazdym wywołaniu tej metody; wydaje mi sie ze mozna zmienic by Mesh extendował Graph i po prostu to był jakiś konstruktor który zwracałby obiekt typu Graph (w javie są kowariancje?)